lightning/src/ln/channelmanager.rs

```diff
@@ -109,71 +109,119 @@ use crate::ln::script::ShutdownScript;
+	///
+	/// Note that at this point, we have not checked that the invoice being paid was actually
+	/// generated by us, but rather it's claiming to pay an invoice of ours.
 	Receive {
-		/// Payment secret and total msat received.
+		/// Information about the amount the sender intended to pay and (potential) proof that this
+		/// is a payment for an invoice we generated, which is also used to associate MPP parts of
+		/// a larger payment.
 		payment_data: msgs::FinalOnionHopData,
-		/// See [`RecipientOnionFields::payment_metadata`] for more info.
+		/// Additional data which we (allegedly) instructed the sender to include in the onion.
+		///
+		/// For HTLCs received by LDK, this will ultimately bubble back up as
+		/// [`RecipientOnionFields::payment_metadata`].
 		payment_metadata: Option<Vec<u8>>,
 		/// CLTV expiry of the received HTLC.
+		///
 		/// Used to track when we should expire pending HTLCs that go unclaimed.
 		incoming_cltv_expiry: u32,
-		/// Shared secret derived using a phantom node secret key. If this field is Some, the
-		/// payment was sent to a phantom node (one hop beyond the current node), but can be
-		/// settled by this node.
+		/// If the onion had forwarding instructions to one of our phantom node SCIDs, this will
+		/// provide the onion shared secret used to decrypt the second level of forwarding
+		/// instructions.
 		phantom_shared_secret: Option<[u8; 32]>,
-		/// See [`RecipientOnionFields::custom_tlvs`] for more info.
+		/// Custom TLVs which were set by the sender.
+		///
+		/// For HTLCs received by LDK, these will ultimately bubble back up as
+		/// [`RecipientOnionFields::custom_tlvs`].
 		custom_tlvs: Vec<(u64, Vec<u8>)>,
 	},
-	/// Incoming keysend (sender provided the preimage in a TLV).
+	/// The onion indicates that this is for payment to us but which contains the preimage for
+	/// claiming included, and is unrelated to any invoice we'd previously generated (aka a
+	/// "keysend" or "spontaneous" payment).
 	ReceiveKeysend {
-		/// This was added in 0.0.116 and will break deserialization on downgrades.
+		/// Information about the amount the sender intended to pay and possibly a token to
+		/// associate MPP parts of a larger payment.
+		///
+		/// This will only be filled in if receiving MPP keysend payments is enabled, and it being
+		/// present may break downgrades to versions of LDK prior to 0.0.116.
 		payment_data: Option<msgs::FinalOnionHopData>,
 		/// Preimage for this onion payment. This preimage is provided by the sender and will be
 		/// used to settle the spontaneous payment.
 		payment_preimage: PaymentPreimage,
-		/// See [`RecipientOnionFields::payment_metadata`] for more info.
+		/// Additional data which we (allegedly) instructed the sender to include in the onion.
+		///
+		/// For HTLCs received by LDK, this will ultimately bubble back up as
+		/// [`RecipientOnionFields::payment_metadata`].
 		payment_metadata: Option<Vec<u8>>,
 		/// CLTV expiry of the received HTLC.
```

Comment:
nit: Adding a space to maintain consistency with the above docs.

```suggestion
		/// CLTV expiry of the received HTLC.
		
```